File: benchmark/fib/fib.pl
use strict;
use warnings;

sub fib {
    my ($n) = @_;
    return $n if $n < 2;
    return fib($n - 1) + fib($n - 2);
}

my $total = 0;
for my $rep (1 .. 40) {
    $total += fib(30);
}
print "$total\n";

File: benchmark/fib/fib.php
<?php

function fib(int $n): int
{
    if ($n < 2) {
        return $n;
    }
    return fib($n - 1) + fib($n - 2);
}

$total = 0;
for ($rep = 0; $rep < 40; $rep += 1) {
    $total += fib(30);
}
echo $total, "\n";
